pinot-core/src/main/java/org/apache/pinot/core/transport/ServerChannels.java

```diff
@@ -151,7 +153,24 @@ private class ServerChannel {
 
     ServerChannel(ServerRoutingInstance serverRoutingInstance) {
       _serverRoutingInstance = serverRoutingInstance;
+      PooledByteBufAllocator bufAllocator = PooledByteBufAllocator.DEFAULT;
+      PooledByteBufAllocatorMetric metric = bufAllocator.metric();
+      _brokerMetrics.setOrUpdateGauge(BrokerGauge.NETTY_POOLED_USED_DIRECT_MEMORY.getGaugeName(),
```

Comment:
So these will be emitted when a channel is established ? How do we capture the change in these memory metrics as a result of queries ?

Reply:
You are right. I wanted to use the suppliers there. Fixed here and in the broker.